### PR TITLE
SpeakerClustering: fix tensor dimennsions in forward()

### PR DESCRIPTION
**Pull request by virajkarandikar**
For some paramters in SpeakerClustering.forward(), tensor dimensions were not specified and this resulted in zero dimension tensor inputs. This causes error when inferencing exported model in Triton server.

# What does this PR do ?

Avoids zero dimension input tensors. Also extracts values from the tensors which have single dim of size 1. This avoids scattered use of item() to extract value from tensor.

**Collection**: ASR Diarizer

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
Fix is needed to use exported clustering model in Triton

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
- [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] Bugfix

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by virajkarandikar** [comment]
@tango4j can you review this?

**Comment by tango4j** [comment]
@virajkarandikar The change looks good. I have added unit-test for jit script exporting. However, this can only check problems on the python side, not for triton server cases.

File: tests/collections/asr/test_diar_utils.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            multiscale_segment_counts[scale_idx] += emb.shape[0]
+
+    emb_tensor = torch.concat(emb_list)
+    multiscale_segment_counts = torch.tensor(multiscale_segment_counts)
+    segm_tensor = torch.tensor(seg_list)
+    multiscale_weights = torch.ones(len(ms_window)).unsqueeze(0)
+    return emb_tensor, segm_tensor, multiscale_segment_counts, multiscale_weights, spk_timestamps
+
+
+@pytest.mark.skip("test is not intended")
```
Comment (nithinraok) [inline tests/collections/asr/test_diar_utils.py:81]: why are we skipping this?

Comment (tango4j) [reply tests/collections/asr/test_diar_utils.py:81]: This function is sub-function for other testcases. This function was included in the test hogging up the time so skipped it.

**Comment by virajkarandikar** [comment]
@tango4j I have changed the forward to use Dict. This allows triton inference by using parameter names instead of INPUT_n notation. See https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md#special-conventions-for-pytorch-backend for details.

**Comment by tango4j** [comment]
@virajkarandikar The cluster inference function should have parameters for re-usability. so I made a wrapper function for the part you created (`forward`). I made nemo users to use `forward_infer` to have parameters in the function call. I think this will work out but please let me know if this does not compile

**Comment by virajkarandikar** [comment]
Ok. It works fine.
Can you add test for forward() function also?

**Comment by tango4j** [comment]
@virajkarandikar Sure, I will add a test for forward().

**Comment by tango4j** [comment]
@virajkarandikar I have added the test-case for .forward() (triton env inference). @nithinraok Please approve if there is no issues with this PR.

**Review by nithinraok — APPROVED** [review]
